sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -40,9 +42,22 @@
 import static com.azure.core.util.FluxUtil.monoError;
 import static com.azure.core.util.FluxUtil.withContext;
 
-/** Initializes a new instance of the asynchronous container repository client.
- * This client is used for interacting with the repository.
- * */
+/**
+ * This class provides a client that contains all the operations for repositories in Azure Container Registry.
+ * Operations allowed by the client are listing, retrieving, deleting, setting writeable properties.
+ * These operations are supported on the repository and the respective tags and manifests in it.
+ *
+ * Listing operations on tags also support {@link ListTagsOptions options}
+ * Listing operations on artifacts also support {@link ListRegistryArtifactOptions options.}
```

Comment:
Not sure we need to call out specific handling for an API on the class level documentation